packages/core/src/Radio.js

```diff
@@ -45,21 +46,31 @@ const RadioInput = styled.input`
   }
 `
 
-const RadioIcon = styled(Icon)`
+const RadioCheckedIcon = styled(RadioChecked)`
   vertical-align: middle;
 `
+const RadioEmptyIcon = styled(RadioEmpty)`
+  vertical-align: middle;
+`
+const getRadioIcon = ({ checked, ...props }) => {
+  return checked ? (
+    <RadioCheckedIcon {...props} />
+  ) : (
+    <RadioEmptyIcon {...props} />
+  )
+}
 
 const Radio = props => {
   const { checked, disabled, size } = props
 
-  const radioIconName = checked ? 'radioChecked' : 'radioEmpty'
-
   const borderAdjustedSize = size + 4
 
+  const radioIcon = getRadioIcon({ checked, size: borderAdjustedSize })
+
   return (
     <RadioWrap color={props.color} checked={checked} disabled={disabled}>
       <RadioInput type="radio" {...props} />
-      <RadioIcon name={radioIconName} size={borderAdjustedSize} />
+      {radioIcon}
```

Comment:
`getRadioIcon` could just be a component

Reply:
@achoboter I pushed this change.